### PR TITLE
chore: Adding semantic release config file

### PR DESCRIPTION
**Pull request by josecorella**
*Description of changes:*
In order to automatically version and publish changelogs we will be using [semantic-release](https://semantic-release.gitbook.io/semantic-release/) to ensure a consistent versioning policy. This file adds a configuration to track [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) both in the commit analyzer and in the changelog format.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

# Check any applicable:
- [ ] Were any files moved? Moving files changes their URL, which breaks all hyperlinks to the files.

### CODE REVIEW

**Comment by texastony** [comment]
> In order to automatically version and publish changelogs we will be using semantic-release ~~to make our lives a lot easier~~ *to ensure a consistent versioning policy*. This file adds a configuration to track conventional commits both in the commit analyzer ~~but~~ *and* in the changelog format.

Nit: Changes to PR description

File: .releaserc [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+            "presetConfig": {
+                "types": [
+                    {"type": "feat", "section": "Features"},
+                    {"type": "fix", "section": "Bug Fixes"},
+                    {"type": "chore", "section": "Maintenance"},
+                    {"type": "docs", "section": "Maintenance"},
+                    {"type": "style", "hidden": true},
+                    {"type": "refactor", "hidden": true},
+                    {"type": "perf", "hidden": true},
+                    {"type": "test", "hidden": true}
```
Comment (texastony) [inline .releaserc:18]: What about revert?

**Review by texastony — CHANGES_REQUESTED** [review]
Add revert and correct PR description. 
Also, what parses `.releaserc`? A Maven plugin? Can you reference the plugin in the PR description?

Sorry if I am being picky... I just don't want to be guessing.

**Comment by josecorella** [comment]
> Add revert and correct PR description.
> Also, what parses `.releaserc`? A Maven plugin? Can you reference the plugin in the PR description?
> 
> Sorry if I am being picky... I just don't want to be guessing.

No worries these are great questions! [semantic-release](https://semantic-release.gitbook.io/semantic-release/) uses the .releaserc file in order to version and update changelogs

**Review by texastony — DISMISSED** [review]
LGTM.

Not dead sure if "Bug Fixes" is correct for revert... but also not sure on what a better term is. If I see one in the wild, I'll cut a PR/share it with you and see what you think.

**Comment by josecorella** [comment]
> LGTM.
> 
> Not dead sure if "Bug Fixes" is correct for revert... but also not sure on what a better term is. If I see one in the wild, I'll cut a PR/share it with you and see what you think.

yeah i was thinking of one too but saw that verdaccio for js-esdk used Bug Fixes for when we [reverted the last commit ](https://github.com/aws/aws-encryption-sdk-javascript/blob/master/CHANGELOG.md#303-2021-09-21)

**Review by texastony — APPROVED** [review]
LGTM: I assume we do not need a package.json... which is news to me.. but I am down with it.

**Pull request merged**